### PR TITLE
implement a dynamic calculation queue name

### PR DESCRIPTION
**Pull request by mokevnin**

### CODE REVIEW

File: activejob/lib/active_job/queue_name.rb [hunk 1]
```diff
@@ -29,22 +29,35 @@ def queue_name_from_part(part_name) #:nodoc:
         name_parts = [queue_name_prefix.presence, queue_name]
         name_parts.compact.join(queue_name_delimiter)
       end
+
+      def queue_name=(v)
+        @queue_name = v
+      end
+
+      def queue_name
+        name = defined?(@queue_name) ? instance_variable_get(:@queue_name) : default_queue_name
```
Comment (egilburg) [inline activejob/lib/active_job/queue_name.rb]: why not just `name = defined?(@queue_name) ? @queue_name : default_queue_name`

Comment (egilburg) [inline activejob/lib/active_job/queue_name.rb]: Also, unless you specifically want to handle user-defined `nil` or `false` for `@queue_name`, you can just do:

def queue_name
  @queue_name || default_queue_name
end

Comment (mokevnin) [inline activejob/lib/active_job/queue_name.rb]: oops), i will remove `instance_variable_get`

but without `defined?` i see:
`warning: instance variable @queue_name not initialized`

**Comment by PlugIN73** [comment]
:+1:

**Comment by kamipo** [comment]
Closing in favor of #34376

**Pull request closed without merging**